File: C#Development/C#OOPBasics/ DefiningClasses-Exercise/06.CompanyRoster/06.CompanyRoster/Employee.cs
﻿namespace _06.CompanyRoster
{
    public class Employee
    {
        private string department;
        private string email;
        private int age;

        public Employee()
        {
        }

        public Employee(string name, decimal salary, string position, string department)
        {
            this.Name = name;
            this.Salary = salary;
            this.Position = position;
            this.Department = department;
        }

        public Employee(string name, decimal salary, string position, string department, string email, int age)
            : this(name, salary, position, department)
        {
            this.Email = email;
            this.Age = age;
        }

        public string Name { get; set; }

        public decimal Salary { get; set; }

        public string Position { get; set; }

        public string Department { get; set; }

        public string Email { get; set; }

        public int Age { get; set; }

        public override string ToString() => $"{this.Name} {this.Salary:F2} {this.Email} {this.Age}";
    }
}

File: JavaScriptCore/JavaScriptWeb/IntroToAngularAndTypeScript/03.People/employee.ts
namespace People {
    export abstract class Employee {
        name: string;
        age: number;
        salary: number;
        tasks: Array<string>;

        constructor(name: string, age: number) {
            this.name = name;
            this.age = age;
            this.salary = 0;
            this.tasks = [];
        }

        work(): void {
            let currentTask: string = this.tasks.shift();
            console.log(this.name + currentTask);
            this.tasks.push(currentTask);
        }

        getSalary(): number {
            return this.salary;
        }

        collectSalary(): void {
            console.log(`${this.name} received ${this.getSalary()} this month.`);
        }
    }
}